File: app/src/main/java/com/funnyvo/android/settings/SettingFragment.java
package com.funnyvo.android.settings;


import android.content.Context;
import android.content.Intent;
import android.content.SharedPreferences;
import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentTransaction;

import com.funnyvo.android.accounts.RequestVerificationFragment;
import com.funnyvo.android.main_menu.MainMenuActivity;
import com.funnyvo.android.main_menu.relatetofragment_onback.RootFragment;
import com.funnyvo.android.R;
import com.funnyvo.android.simpleclasses.ApiRequest;
import com.funnyvo.android.simpleclasses.Callback;
import com.funnyvo.android.simpleclasses.Variables;
import com.funnyvo.android.simpleclasses.WebviewFragment;

import org.json.JSONException;
import org.json.JSONObject;

import static com.funnyvo.android.simpleclasses.Variables.API_SUCCESS_CODE;

/**
 * A simple {@link Fragment} subclass.
 */
public class SettingFragment extends RootFragment implements View.OnClickListener {

    private View view;
    private Context context;

    public SettingFragment() {
        // Required empty public constructor
    }

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        // Inflate the layout for this fragment
        view = inflater.inflate(R.layout.fragment_setting, container, false);
        context = getContext();

        return view;
    }

    @Override
    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);
        view.findViewById(R.id.Goback).setOnClickListener(this);
        view.findViewById(R.id.request_verification_txt).setOnClickListener(this);
        view.findViewById(R.id.privacy_policy_txt).setOnClickListener(this);
        view.findViewById(R.id.logout_txt).setOnClickListener(this);
    }

    @Override
    public void onClick(View v) {
        switch (v.getId()) {

            case R.id.Goback:
                getActivity().onBackPressed();
                break;

            case R.id.request_verification_txt:
                openRequestVerification();
                break;

            case R.id.privacy_policy_txt:
                openPrivacyUrl();
                break;

            case R.id.logout_txt:
                logout();
                break;
        }
    }


    private void openRequestVerification() {
        RequestVerificationFragment request_verificationFragment = new RequestVerificationFragment();
        FragmentTransaction transaction = getActivity().getSupportFragmentManager().beginTransaction();
        transaction.setCustomAnimations(R.anim.in_from_right, R.anim.out_to_left, R.anim.in_from_left, R.anim.out_to_right);
        transaction.addToBackStack(null);
        transaction.replace(R.id.Setting_F, request_verificationFragment).commit();
    }

    private void openPrivacyUrl() {
        WebviewFragment webview_fragment = new WebviewFragment();
        FragmentTransaction transaction = getActivity().getSupportFragmentManager().beginTransaction();
        transaction.setCustomAnimations(R.anim.in_from_right, R.anim.out_to_left, R.anim.in_from_left, R.anim.out_to_right);
        Bundle bundle = new Bundle();
        bundle.putString("url", Variables.privacy_policy);
        bundle.putString("title", "Privacy Policy");
        webview_fragment.setArguments(bundle);
        transaction.addToBackStack(null);
        transaction.replace(R.id.Setting_F, webview_fragment).commit();
    }

    // this will erase all the user info store in locally and logout the user
    private void logout() {
        JSONObject parameters = new JSONObject();
        try {
            parameters.put("fb_id", Variables.sharedPreferences.getString(Variables.u_id, ""));
        } catch (JSONException e) {
            e.printStackTrace();
        }

        ApiRequest.callApi(getActivity(), Variables.Logout, parameters, new Callback() {
            @Override
            public void response(String response) {
                try {
                    JSONObject jsonObject = new JSONObject(response);
                    String code = jsonObject.optString("code");
                    if (code.equals(API_SUCCESS_CODE)) {
                        SharedPreferences.Editor editor = Variables.sharedPreferences.edit();
                        editor.putString(Variables.u_id, "").clear();
                        editor.putString(Variables.u_name, "").clear();
                        editor.putString(Variables.u_pic, "").clear();
                        editor.putBoolean(Variables.islogin, false).clear();
                        editor.apply();
                        getActivity().finish();
                        startActivity(new Intent(getActivity(), MainMenuActivity.class));
                    }
                } catch (JSONException e) {
                    e.printStackTrace();
                }

            }
        });
    }


}

File: app/src/main/java/com/funnyvo/android/settings/SettingFragment.kt
package com.funnyvo.android.settings

import android.content.Intent
import android.net.Uri
import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ArrayAdapter
import com.funnyvo.android.BuildConfig
import com.funnyvo.android.R
import com.funnyvo.android.accounts.RequestVerificationFragment
import com.funnyvo.android.main_menu.MainMenuActivity
import com.funnyvo.android.main_menu.relatetofragment_onback.RootFragment
import com.funnyvo.android.simpleclasses.ApiRequest
import com.funnyvo.android.simpleclasses.Variables
import com.funnyvo.android.simpleclasses.WebviewFragment
import com.funnyvo.android.splash.SplashActivity
import kotlinx.android.synthetic.main.fragment_setting.*
import org.json.JSONException
import org.json.JSONObject


/**
 * A simple [Fragment] subclass.
 */
class SettingFragment : RootFragment(), View.OnClickListener {

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?,
                              savedInstanceState: Bundle?): View? {
        // Inflate the layout for this fragment
        return inflater.inflate(R.layout.fragment_setting, container, false)
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        btnGoBackSettings.setOnClickListener(this)
        request_verification_txt.setOnClickListener(this)
        privacy_policy_txt.setOnClickListener(this)
        logout_txt.setOnClickListener(this)
        txtRateApp.setOnClickListener(this)
        val items = listOf("Everyone", "Followers", "Private")
        val adapter = ArrayAdapter(requireContext(), R.layout.view_selection_dropdown, items)
        txtVideoVisibleDropdown.setAdapter(adapter)
//        txtVideoVisibleDropdown.postDelayed(Runnable {
//            txtVideoVisibleDropdown.setText(Variables.sharedPreferences.getString(Variables.USER_PREF_VIDEO_VISIBILITY, "Everyone"))
//            txtVideoVisibleDropdown.showDropDown()
//        }, 10)
        txtVideoVisibleDropdown.hint = Variables.sharedPreferences.getString(Variables.USER_PREF_VIDEO_VISIBILITY, "Everyone")

        txtVideoVisibleDropdown.setOnItemClickListener { adapterView, view, position, l ->
            val selection: String = adapterView.getItemAtPosition(position) as String
            val editor = Variables.sharedPreferences.edit()
            editor.putString(Variables.USER_PREF_VIDEO_VISIBILITY, selection)
            editor.apply()
        }
    }

    override fun onClick(v: View) {
        when (v.id) {
            R.id.btnGoBackSettings -> activity?.onBackPressed()
            R.id.request_verification_txt -> openRequestVerification()
            R.id.privacy_policy_txt -> openPrivacyUrl()
            R.id.logout_txt -> logout()
            R.id.txtRateApp -> openAppPlayStore()
        }
    }

    private fun openAppPlayStore() {
        val intent = Intent(Intent.ACTION_VIEW, Uri.parse("market://details?id=" + BuildConfig.APPLICATION_ID))
        startActivity(intent)
    }

    private fun openRequestVerification() {
        val requestVerificationFragment = RequestVerificationFragment()
        val transaction = requireActivity().supportFragmentManager.beginTransaction()
        transaction.setCustomAnimations(R.anim.in_from_right, R.anim.out_to_left, R.anim.in_from_left, R.anim.out_to_right)
        transaction.addToBackStack(null)
        transaction.replace(R.id.settingsFrameLayout, requestVerificationFragment).commit()
    }

    private fun openPrivacyUrl() {
        val webviewFragment = WebviewFragment()
        val transaction = requireActivity().supportFragmentManager.beginTransaction()
        transaction.setCustomAnimations(R.anim.in_from_right, R.anim.out_to_left, R.anim.in_from_left, R.anim.out_to_right)
        val bundle = Bundle()
        bundle.putString("url", Variables.privacy_policy)
        bundle.putString("title", "Privacy Policy")
        webviewFragment.arguments = bundle
        transaction.addToBackStack(null)
        transaction.replace(R.id.settingsFrameLayout, webviewFragment).commit()
    }

    // this will erase all the user info store in locally and logout the user
    private fun logout() {
        val parameters = JSONObject()
        try {
            parameters.put("fb_id", Variables.sharedPreferences.getString(Variables.u_id, ""))
        } catch (e: JSONException) {
            e.printStackTrace()
        }
        ApiRequest.callApi(activity, Variables.LOGOUT, parameters) { response ->
            try {
                if(response.isNotEmpty()) {
                    val jsonObject = JSONObject(response)
                    val code = jsonObject.optString("code")
                    if (code == Variables.API_SUCCESS_CODE) {
                        val editor = Variables.sharedPreferences.edit()
                        editor.putString(Variables.u_id, "").clear()
                        editor.putString(Variables.u_name, "").clear()
                        editor.putString(Variables.u_pic, "").clear()
                        editor.putBoolean(Variables.islogin, false).clear()
                        editor.apply()
                        val intent = Intent(activity, SplashActivity::class.java)
                        intent.flags = Intent.FLAG_ACTIVITY_CLEAR_TASK or Intent.FLAG_ACTIVITY_CLEAR_TOP
                        startActivity(intent)
                        requireActivity().finish()
                    }
                }
            } catch (e: JSONException) {
                e.printStackTrace()
            }
        }
    }
}